### PR TITLE
change function signature to get type name via intrinsics

### LINKED ISSUE #18354: Replace manual implementations of Runnable::name with automatic type name
We have a [Runnable](https://doc.servo.org/script/script_thread/trait.Runnable.html) trait with a name method. This method is used to improve profiling output so we can tell at runtime which concrete runnable is being executed, so we have to duplicate the name of the type every time we create a new type that implements Runnable. We can remove all of that if we use this code for Runnable::name instead: https://gist.github.com/2210a8674dd7813cec8c7148a3a0b719

Code: `components/script/script_thread.rs`, and every implementation of Runnable that overrides the name method.

No need to run any tests for this change; if it builds, that's good enough for a PR.

### PR DESCRIPTION
**Pull request by toidiu**
change function signature to get type name via intrinsics

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #18354 (github issue number if applicable).

- [ ] There are tests for these changes OR
- [x] These changes do not require tests because stated in the PR that it was not required

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/18361)

### CODE REVIEW

File: components/script/script_thread.rs [hunk 1]
```diff
@@ -221,7 +222,7 @@ pub struct CancellableRunnable<T: Runnable + Send> {
 }
 
 impl<T: Runnable + Send> Runnable for CancellableRunnable<T> {
-    fn name(&self) -> &'static str { self.inner.name() }
+    fn name(&self) -> &'static str { unsafe { intrinsics::type_name::<Self>() } }
```
Comment (emilio) [inline components/script/script_thread.rs]: I believe this should be in the definition of `trait Runnable`, and after that all the manual implementations should go away.

Comment (toidiu) [reply components/script/script_thread.rs]: makes sense. will implement the change

**Comment by toidiu** [comment]
@emilio temporarily removed `name()` to let the compiled find all instances of it and then removed them. This should be ready for review.

**Review by emilio — APPROVED** [review]
Ok, this patch makes much more sense to me, thanks!

**Pull request merged**